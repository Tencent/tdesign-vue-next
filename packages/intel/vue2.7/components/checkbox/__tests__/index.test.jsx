import { mount } from '@vue/test-utils';
import Checkbox, { CheckboxGroup } from '@/src/checkbox/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Checkbox', () => {
  // test props api
  describe(':props', () => {
    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <Checkbox checked={true}></Checkbox>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':defaultChecked', () => {
      const wrapper = mount({
        render() {
          return <Checkbox defaultChecked={true}></Checkbox>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <Checkbox disabled={true}></Checkbox>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':indeterminate', () => {
      const wrapper = mount({
        render() {
          return <Checkbox indeterminate={true}></Checkbox>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':name', () => {
      const wrapper = mount({
        render() {
          return <Checkbox name={'checkbox-name'}></Checkbox>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('Event passthrough', async () => {
      const wrapper = mount(Checkbox);
      const checkboxInput = wrapper.find('input[type="checkbox"]');
      await checkboxInput.setChecked();
      expect(checkboxInput.element.checked).toBeTruthy();

      // TODO emitted return undefined in vue 2.7
      // expect(wrapper.emitted().change.length).toBe(1);
      // expect(wrapper.emitted().change[0][0]).toBeTruthy();
    });
  });
});

describe('Checkbox CheckboxGroup', () => {
  // test props api
  describe(':props', () => {
    it(':defaultChecked', () => {
      const wrapper = mount({
        render() {
          return (
            <CheckboxGroup defaultChecked={['sz']}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>
                深圳
              </Checkbox>
            </CheckboxGroup>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':checked', () => {
      const wrapper = mount({
        render() {
          return (
            <CheckboxGroup checked={['sz']}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>
                深圳
              </Checkbox>
            </CheckboxGroup>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return (
            <CheckboxGroup disabled={true}>
              <Checkbox value="bj">北京</Checkbox>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled={false}>
                深圳
              </Checkbox>
            </CheckboxGroup>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':max', () => {
      const options = [
        { value: 'bj', label: '北京' },
        { value: 'gz', label: '广州' },
        { value: 'sz', label: '深圳' },
      ];
      const wrapper = mount({
        render() {
          return <CheckboxGroup options={options} value={['sz', 'gz']} max={2} />;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':options', () => {
      const options = [
        { value: 'gz', label: '广州', disabled: true },
        { value: 'sz', label: '深圳' },
      ];
      const wrapper = mount({
        render() {
          return <CheckboxGroup options={options}></CheckboxGroup>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':name', () => {
      const wrapper = mount({
        render() {
          return (
            <CheckboxGroup name={'checkbox-name'}>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>
                深圳
              </Checkbox>
            </CheckboxGroup>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('Event passthrough', async () => {
      const wrapper = mount({
        render() {
          return (
            <CheckboxGroup>
              <Checkbox value="gz">广州</Checkbox>
              <Checkbox value="sz" disabled>
                深圳
              </Checkbox>
            </CheckboxGroup>
          );
        },
      });
      const checkboxInput = wrapper.find('input[type="checkbox"]');
      await checkboxInput.setChecked();
      // const checkboxGroup = wrapper.findComponent(CheckboxGroup);
      // const checkbox = wrapper.findComponent(Checkbox);
      // expect(checkbox.emitted().change.length).toBe(1);
      // expect(checkboxGroup.emitted().change.length).toBe(1);
      // 选中单项时间参数为 true
      // expect(checkbox.emitted().change[0][0]).toBeTruthy();
      // checkboxGroup 事件参数为 ['gz']
      // expect(checkboxGroup.emitted().change[0][0]).toEqual(['gz']);
    });
  });
});
