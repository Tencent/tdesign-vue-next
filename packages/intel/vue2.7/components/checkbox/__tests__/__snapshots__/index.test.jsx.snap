// Vitest Snapshot v1

exports[`Checkbox > :props > :checked 1`] = `
<label
  class="t-checkbox t-is-checked"
  tabindex="0"
>
  <input
    class="t-checkbox__former"
    tabindex="-1"
    type="checkbox"
    value=""
  />
  <span
    class="t-checkbox__input"
  />
  <span
    class="t-checkbox__label"
  />
</label>
`;

exports[`Checkbox > :props > :defaultChecked 1`] = `
<label
  class="t-checkbox t-is-checked"
  tabindex="0"
>
  <input
    class="t-checkbox__former"
    tabindex="-1"
    type="checkbox"
    value=""
  />
  <span
    class="t-checkbox__input"
  />
  <span
    class="t-checkbox__label"
  />
</label>
`;

exports[`Checkbox > :props > :disabled 1`] = `
<label
  class="t-checkbox t-is-disabled"
>
  <input
    class="t-checkbox__former"
    disabled="disabled"
    tabindex="-1"
    type="checkbox"
    value=""
  />
  <span
    class="t-checkbox__input"
  />
  <span
    class="t-checkbox__label"
  />
</label>
`;

exports[`Checkbox > :props > :indeterminate 1`] = `
<label
  class="t-checkbox t-is-indeterminate"
  tabindex="0"
>
  <input
    class="t-checkbox__former"
    indeterminate="indeterminate"
    tabindex="-1"
    type="checkbox"
    value=""
  />
  <span
    class="t-checkbox__input"
  />
  <span
    class="t-checkbox__label"
  />
</label>
`;

exports[`Checkbox > :props > :name 1`] = `
<label
  class="t-checkbox"
  tabindex="0"
>
  <input
    class="t-checkbox__former"
    name="checkbox-name"
    tabindex="-1"
    type="checkbox"
    value=""
  />
  <span
    class="t-checkbox__input"
  />
  <span
    class="t-checkbox__label"
  />
</label>
`;

exports[`Checkbox CheckboxGroup > :props > :checked 1`] = `
<div
  aria-label="checkbox-group"
  checked="checked"
  class="t-checkbox-group"
  role="group"
>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled="disabled"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;

exports[`Checkbox CheckboxGroup > :props > :defaultChecked 1`] = `
<div
  aria-label="checkbox-group"
  class="t-checkbox-group"
  defaultchecked="sz"
  role="group"
>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled="disabled"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;

exports[`Checkbox CheckboxGroup > :props > :disabled 1`] = `
<div
  aria-label="checkbox-group"
  class="t-checkbox-group"
  role="group"
>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="bj"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      北京
    </span>
  </label>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;

exports[`Checkbox CheckboxGroup > :props > :max 1`] = `
<div
  aria-label="checkbox-group"
  class="t-checkbox-group"
  role="group"
>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="bj"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      北京
    </span>
  </label>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;

exports[`Checkbox CheckboxGroup > :props > :name 1`] = `
<div
  aria-label="checkbox-group"
  class="t-checkbox-group"
  role="group"
>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled="disabled"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;

exports[`Checkbox CheckboxGroup > :props > :options 1`] = `
<div
  aria-label="checkbox-group"
  class="t-checkbox-group"
  role="group"
>
  <label
    class="t-checkbox t-is-disabled"
  >
    <input
      class="t-checkbox__former"
      disabled="disabled"
      tabindex="-1"
      type="checkbox"
      value="gz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      广州
    </span>
  </label>
  <label
    class="t-checkbox"
    tabindex="0"
  >
    <input
      class="t-checkbox__former"
      tabindex="-1"
      type="checkbox"
      value="sz"
    />
    <span
      class="t-checkbox__input"
    />
    <span
      class="t-checkbox__label"
    >
      深圳
    </span>
  </label>
</div>
`;
